First iteration of converting files to typescript

aadnekar opened the pull request.

aadnekar commented on src/components/Chart/data/dataSelector.ts:
```diff
@@ -8,17 +8,20 @@
 
-    process(begin, end, digitalChannelsToCompute) {
+    process(
+        begin: number,
+        end: number,
```
@datenreisender , I think you mentioned that it would be nice if this were either typed with index or timestamp according to if they are index or type, since this isn't always obvious. Therefore, I wonder if we should have created specific types for e.g. `sampleIndex` and `sampleTimestamp`, which are basically `number` type?

datenreisender replied:
Yes. Maybe also make types for things like a rate and a duration in microseconds, so it is something like this:

```ts
type rate = number;
type microseconds = number;
type sampleIndex = number;
type timestamp = number;

interface GlobalOptions {
    samplingTime: microseconds;
    samplesPerSecond: rate;
    currentIndex: sampleIndex;
    currentTimestamp: timestamp | undefined | null;
}
```

These types sometimes get lost and then TypeScript again tells you that it is just a `number` and it doesn't catch any type errors (you can still say e.g. `options.samplingTime = options.samplesPerSecond`) but it still gives you some hints about the intention.

aadnekar commented:
Will make this an iterative process, and for now I've implemented some of the feedback that I have received:

- [x] Added types to `globals.ts` in order to get types in all other files, this will hopefully be very useful since even though most files are still `.js`, vs code still presents the types if they are imported from the `.ts` file.
- [x] `setSamplingRates` changed to `setSamplingRate`
- [x] Fixed a misunderstanding of the type of `digitalChannelsToCompute`
- [x] Loosened types specificity a bit in order to make the refactoring a little easier, and will tighten types iteratively.

aadnekar commented:
Had to force push unchanged amend twice in order to trigger license/cla check.
